### ISSUE TITLE
Syntax checker not checking argument count?

### ISSUE THREAD

**Issue by TheExDeus** [enigma-dev/enigma-dev#931]
When I type draw_sprite(spr,-1,mouse_x.mouse_y); it doesn't trow the "too few arguments" error until the compile phase. The parser should probably catch that. I then tried draw_sprite(spr,-1,mouse_x), draw_sprite(spr) or even draw_sprite() and it doesn't show any error. I guarantee it worked before, so it must be some kind of regression. It doesn't work for any function I tried. The only errors it returns for now are missing functions, brace errors and parenthesis errors.

**Comment by RobertBColton** [comment]
Ok so there's a couple of things here.

* It did used to work but I have not decided who broke it.
* The file we used to use for syntax checking was https://github.com/enigma-dev/enigma-dev/blob/d9746dfc6d557c4c22c8bd8cf24e02e3272eade0/CompilerSource/syntax/chkfunction.h
* The file that has the new syntax check we still use is https://github.com/enigma-dev/enigma-dev/blob/d9746dfc6d557c4c22c8bd8cf24e02e3272eade0/CompilerSource/syntax/syntax.cpp
* We could probably delete the old file since it doesn't handle nearly as much and it's not needed
* The new file has a lot of redundant white space like spaces on empty lines
* If you hit the syntax checking button on an empty script it will infinite loop so the new file/syntax check function needs to check code.empty()
* Finally, why is the new function called syntacheck and not syntaxcheck?

**Comment by RobertBColton** [comment]
Ok so I've wittled the problem down some more.

When I add the following to https://github.com/enigma-dev/enigma-dev/blob/master/CompilerSource/syntax/syntax.cpp#L636
```cpp
syerr = "Dump " + tostring(params) + " " + tostring(minarg);
            return 0;
```

minarg is showing as 0. So it's saying 1 argument was supplied when I do ``draw_sprite(0);`` and 0 arguments are required. So the issue is in the call to ``definition_parameter_bounds`` or its first argument.

**Comment by RobertBColton** [comment]
Closing as fully resolved in #1016
